src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java

```diff
@@ -207,19 +204,18 @@ public AnomalyDetectionIndices(
         this.allSettingUpdated = false;
         this.updateRunning = new AtomicBoolean(false);
 
-        Map<Setting<?>, Consumer<?>> settingToConsumerMap = new HashMap<>();
-        settingToConsumerMap.put(AD_RESULT_HISTORY_MAX_DOCS_PER_SHARD, it -> historyMaxDocs = (Long) it);
-        settingToConsumerMap.put(AD_RESULT_HISTORY_ROLLOVER_PERIOD, it -> {
-            historyRolloverPeriod = (TimeValue) it;
-            rescheduleRollover();
-        });
-        settingToConsumerMap.put(AD_RESULT_HISTORY_RETENTION_PERIOD, it -> historyRetentionPeriod = (TimeValue) it);
-        settingToConsumerMap.put(MAX_PRIMARY_SHARDS, it -> maxPrimaryShards = (int) it);
-        try {
-            this.clusterService.getClusterSettings().addSettingsUpdateConsumer(settingToConsumerMap);
-        } catch (Exception e) {
-            // FIXME Handle this
-            // https://github.com/opensearch-project/opensearch-sdk-java/issues/422
+        if (clusterService != null) {
```

Comment:
This is good. I think we should add this check for AD as well [here](https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java#L202-L212) on `main`